python/kserve/kserve/api/kserve_client.py

```diff
@@ -27,21 +27,31 @@
 
 class KServeClient(object):
 
-    def __init__(self, config_file=None, context=None,  # pylint: disable=too-many-arguments
+    def __init__(self, config_file=None, config_dict=None, context=None,  # pylint: disable=too-many-arguments
                  client_configuration=None, persist_config=True):
         """
         KServe client constructor
         :param config_file: kubeconfig file, defaults to ~/.kube/config
+        :param config_dict: Takes the config file as a dict.
         :param context: kubernetes context
         :param client_configuration: kubernetes configuration object
         :param persist_config:
         """
-        if config_file or not utils.is_running_in_k8s():
-            config.load_kube_config(
-                config_file=config_file,
-                context=context,
-                client_configuration=client_configuration,
-                persist_config=persist_config)
+        if config_file or config_dict or not utils.is_running_in_k8s():
```

Comment:
How is config_dict different from client_configuration which also seems to be a dict

Reply:
Both `load_kube_config()` and `load_kube_config_from_dict()` has a param named `client_configuration`, which is a class [kubernetes.client.Configuration](https://github.com/kubernetes-client/python/blob/master/kubernetes/client/configuration.py#L25). It can be used to override the existing configuration object. 
There are ways we can avoid using `load_kube_config()` or `load_kube_config_from_dict()` and just use the class [kubernetes.client.Configuration](https://github.com/kubernetes-client/python/blob/master/kubernetes/client/configuration.py#L25) to make authenticated API requests to the cluster(example below).
```
import kubernetes.client

configuration = kubernetes.client.Configuration()
configuration.ssl_ca_cert = 'ca.crt'
configuration.api_key['authorization'] = 'ZXXXYXISNJNSXXXdw=='
configuration.api_key_prefix['authorization'] = 'Bearer'
configuration.host = 'https://garyasuid8ekjkasd.gr7.eu-west-1.eks.amazonaws.com'

api_instance = kubernetes.client.CoreV1Api(kubernetes.client.ApiClient(configuration))

api_response = api_instance.list_namespace()
for i in api_response.items:
    print(i.metadata.name)
```